### PR TITLE
Sync changes from main subql repo

### PR DESCRIPTION
**Pull request by stwiname**

### CODE REVIEW

File: packages/node/src/indexer/x-provider/http.ts [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+import defaults from '@polkadot/rpc-provider/defaults';
+import { LRUCache } from '@polkadot/rpc-provider/lru';
+import type {
+  JsonRpcResponse,
+  ProviderInterface,
+  ProviderInterfaceCallback,
+  ProviderInterfaceEmitCb,
+  ProviderInterfaceEmitted,
+  ProviderStats,
+} from '@polkadot/rpc-provider/types';
```
Comment (DeveloperInProgress) [inline packages/node/src/indexer/x-provider/http.ts:17]: is this file substrate specific?

File: packages/node/src/utils/sync-helper.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 // Copyright 2020-2022 OnFinality Limited authors & contributors
 // SPDX-License-Identifier: Apache-2.0
 
+import { blake2AsHex } from '@polkadot/util-crypto';
```
Comment (DeveloperInProgress) [inline packages/node/src/utils/sync-helper.ts:4]: This adds a polkadot dependency

File: packages/node/test/projectFixture/template-v1.0.0/distMock/index.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+'use strict';
+Object.defineProperty(exports, '__esModule', { value: true });
+const tslib_1 = require('tslib');
+//Exports all handler functions
+require('@polkadot/api-augment');
+(0, tslib_1.__exportStar)(require('./mappings/mappingHandlers'), exports);
```
Comment (DeveloperInProgress) [inline packages/node/test/projectFixture/template-v1.0.0/distMock/index.js:6]: Looks like these test files are all substrate specific

File: packages/node/src/indexer/indexer.manager.ts [hunk 4]
```diff
@@ -420,7 +420,7 @@ const ProcessorTypeMap = {
 };
 
 const FilterTypeMap = {
-  [SubqlCosmosHandlerKind.Block]: () => true,
+  [SubqlCosmosHandlerKind.Block]: () => CosmosUtil.filterBlock,
```
Comment (DeveloperInProgress) [inline packages/node/src/indexer/indexer.manager.ts:423]:
```suggestion
  [SubqlCosmosHandlerKind.Block]: CosmosUtil.filterBlock,
```

File: packages/node/src/utils/sync-helper.ts (hunk 2 above)
Comment (stwiname) [reply packages/node/src/utils/sync-helper.ts:4]: This is fine, its just crypto utils and we're using some polkadot deps for MMR/POI

**Review by DeveloperInProgress — APPROVED** [review]

**Pull request merged**